RecoEgamma/EgammaTools/plugins/EGExtraInfoModifierFromDB.cc

```diff
@@ -364,118 +399,126 @@ namespace {
 }
 
 void EGExtraInfoModifierFromDB::modifyObject(reco::GsfElectron& ele) const {
+
   // regression calculation needs no additional valuemaps
 
   const reco::SuperClusterRef& the_sc = ele.superCluster();
   const edm::Ptr<reco::CaloCluster>& theseed = the_sc->seed();
   const int numberOfClusters =  the_sc->clusters().size();
   const bool missing_clusters = !the_sc->clusters()[numberOfClusters-1].isAvailable();
-
-
-  //skip HGCal for now
-  if( theseed->seed().det() == DetId::Forward ) return;
+  
   if( missing_clusters ) return ; // do not apply corrections in case of missing info (slimmed MiniAOD electrons)
+
+  const bool iseb = ele.isEB();  
+  edm::Handle<EcalRecHitCollection > ecalRecHits;
+  if (iseb) ecalRecHits = ecalRecHitsEB_ ;
+  else      ecalRecHits = ecalRecHitsEE_ ;
+  if ( !ecalRecHits.isValid() ) return;
   
-  std::array<float, 33> eval;  
+  Int_t N_SATURATEDXTALS  = 0;
+  std::vector< std::pair<DetId, float> > hitsAndFractions = theseed->hitsAndFractions();
+  for (auto hitFractionPair : hitsAndFractions) {    
+    auto ecalRecHit = ecalRecHits->find(hitFractionPair.first);
+    if (ecalRecHit == ecalRecHits->end()) continue;
+    if (ecalRecHit->checkFlag(EcalRecHit::Flags::kSaturated)) N_SATURATEDXTALS++;
+  }
+
+  std::array<float, 40> eval;  
   const double raw_energy = the_sc->rawEnergy(); 
-  const auto& ess = ele.showerShape();
-
-  // SET INPUTS
-  eval[0]  = nVtx_;  
-  eval[1]  = raw_energy;
-  eval[2]  = the_sc->eta();
-  eval[3]  = the_sc->phi();
-  eval[4]  = the_sc->etaWidth();
-  eval[5]  = the_sc->phiWidth(); 
-  eval[6]  = ess.r9;
-  eval[7]  = theseed->energy()/raw_energy;
-  eval[8]  = ess.eMax/raw_energy;
-  eval[9]  = ess.e2nd/raw_energy;
-  eval[10] = (ess.eLeft + ess.eRight != 0.f  ? (ess.eLeft-ess.eRight)/(ess.eLeft+ess.eRight) : 0.f);
-  eval[11] = (ess.eTop  + ess.eBottom != 0.f ? (ess.eTop-ess.eBottom)/(ess.eTop+ess.eBottom) : 0.f);
-  eval[12] = ess.sigmaIetaIeta;
-  eval[13] = ess.sigmaIetaIphi;
-  eval[14] = ess.sigmaIphiIphi;
-  eval[15] = std::max(0,numberOfClusters-1);
-  
+  const double raw_es_energy = the_sc->preshowerEnergy();
+  const auto& full5x5_ess = ele.full5x5_showerShape();
+
+  eval[0]  = raw_energy;
+  eval[1]  = the_sc->etaWidth();
+  eval[2]  = the_sc->phiWidth(); 
+  eval[3]  = full5x5_ess.e5x5/raw_energy;
```

Comment:
Please define ``` rawInverse ``` in  a similar way. It's used multiple times here (in [3] and then in [25-27])